_posts/2021-12-15-hhvm-4.140.markdown

```diff
@@ -0,0 +1,23 @@
+---
+title: "HHVM 4.140"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.140 is released! This release marks the end of support for 4.134; HHVM
+4.135&ndash;4.139 remain supported, as do the 4.102 and 4.128 LTS releases.
+
+# Highlights
+
+- Various bug fixes for the `readonly` data types.
+- Various error message improvements.
+
+
+# Breaking Changes
+
+- `JitAHotSize` and `JitAProfSize` runtime options are removed
+- Ban `UNSAFE_CAST` in lvalues and `await` operands. The following code does not compile any more:
```

Comment:
Hack is a weird language, so compile does not have a clear unambiguous meaning. I think that "typecheck" would be a fine term to use here. If this change affects both the typechecker and the runtime, you can add "and will throw at runtime". If this fails when requiring the file (via include directives or autoloading), "parse" would be a fine way to describe it.

Reply:
This change does not affect runtime. But note even though `HH\FIXME\UNSAFE_CAST<mixed, dict<int, string>>($m)[0] = "lvalue";` does not throw at runtime, it does not change `$m`. The runtime behavior is not defined, but according to my tests, the behavior is currently the same as `$tmp = HH\FIXME\UNSAFE_CAST<mixed, dict<int, string>>($m); $tmp[0] = "lvalue"`.